File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/common/uicomponent/barrage/service/IEmojiResource.java
package com.trtc.uikit.livekit.common.uicomponent.barrage.service;

import android.content.Context;
import android.graphics.Rect;
import android.graphics.drawable.Drawable;

import java.util.List;

public interface IEmojiResource {
    int getResId(String key);

    List<Integer> getResIds();

    String getEncodeValue(int resId);

    String getEncodePattern();

    Drawable getDrawable(Context context, int resId, Rect bounds);
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/component/barrage/viewmodel/IEmojiResource.kt
package com.trtc.uikit.livekit.component.barrage.viewmodel

import android.content.Context
import android.graphics.Rect
import android.graphics.drawable.Drawable

interface IEmojiResource {
    fun getResId(key: String): Int

    fun getResIds(): List<Int>

    fun getEncodeValue(resId: Int): String

    fun getEncodePattern(): String

    fun getDrawable(context: Context, resId: Int, bounds: Rect?): Drawable
}
